ahnlich/ai/src/engine/ai/providers/fastembed.rs

```diff
@@ -182,36 +184,35 @@ impl ProviderTrait for FastEmbedProvider {
         //     TODO (HAKSOAT): When we add model specific tokenizers, add the get tokenizer call here too.
     }
 
-    fn run_inference(
-        &self,
-        input: &ModelInput,
-        action_type: &InputAction,
-    ) -> Result<Vec<f32>, AIProxyError> {
-        if let Some(fastembed_model) = &self.model {
-            let embeddings = match fastembed_model {
-                FastEmbedModel::Text(model) => {
-                    if let ModelInput::Text(value) = input {
-                        model
-                            .embed(vec![value], None)
-                            .map_err(|_| AIProxyError::ModelProviderRunInferenceError)?
-                    } else {
-                        let store_input_type: AIStoreInputType = input.into();
-                        let Some(index_model_repr) = self.supported_models else {
-                            return Err(AIProxyError::AIModelNotInitialized);
-                        };
-                        let index_model_repr: Model = (&index_model_repr).into();
-                        return Err(AIProxyError::StoreTypeMismatchError {
-                            action: *action_type,
-                            index_model_type: index_model_repr.input_type(),
-                            storeinput_type: store_input_type,
-                        });
-                    }
+    fn run_inference(&self, inputs: &[ModelInput], action_type: &InputAction) -> Result<Vec<Vec<f32>>, AIProxyError> {
```

Comment:
Can `run_inference` return the `Vec<StoreKey>`? so that higher up callers no longer have to iterate over the vec just to cast to another type